lld/ELF/Arch/X86_64.cpp

```diff
@@ -897,6 +901,7 @@ RelExpr X86_64::adjustGotPcExpr(RelType type, int64_t addend,
 static void relaxGotNoPic(uint8_t *loc, uint64_t val, uint8_t op,
                           uint8_t modRm) {
   const uint8_t rex = loc[-3];
```

Comment:
The code assumes correctly used relocations to avoid some bounds checking. 
We don't bother with erroneous relocation testing - for valid `R_X86_64_REX_GOTPCRELX`, `loc[-3]` should always be accessible.

However, if type is `R_X86_64_REX_GOTPCRELX`, `loc[-4]` could go out-of-bounds.

Reply:
We can pass `RelType type` or `bool isRex2` here as a parameter, which one do you prefer?

Reply:
Passing in `isRex2` LG

Reply:
Done